.github/workflows/set-milestone.yml

```diff
@@ -0,0 +1,36 @@
+#############################################################################
+# GitHub action to set latest milestone on issue of merged PR.
+#
+#############################################################################
+name: 'Set milestone on issue of merged PR'
+
+on:
+  pull_request:
+    types:
+      - closed
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  set-milestone:
+    name: 'Set milestone on issue of merged PR #${{ github.event.pull_request.number }}'
+    if: ${{ github.event.pull_request.merged == true
+      && (startsWith(github.event.pull_request.title, 'Issue')
+      || startsWith(github.event.pull_request.title, 'Pull')) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the latest code
+        uses: actions/checkout@v3
+
+      - name: Set milestone on issue
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PULL_REQUEST_TITLE: ${{ github.event.pull_request.title }}
+        run: |
+          ./.ci/set-milestone.sh
```

Comment:
we have milestones for release.

lets name this as something like: "set-milestone-on-referenced-issue"

Reply:
Done.